mmv1/third_party/terraform/services/apigee/resource_apigee_api.go

```diff
@@ -0,0 +1,403 @@
+				Description: `A hash of local config bundle in string, user needs to use a Terraform Hash function of their choice. A change in hash will trigger an update.`,
+				DiffSuppressFunc: func(k, old, new string, d *schema.ResourceData) bool {
+					localMd5Hash := ""
+					if config_bundle, ok := d.GetOkExists("config_bundle"); ok {
+						localMd5Hash = tpgresource.GetFileMd5Hash(config_bundle.(string))
+					}
+					if localMd5Hash == "" {
+						return false
+					}
+
+					// `old` is the md5 hash we speculated from server responses,
+					// when apply responded with succeed, hash is set to the hash of uploaded bundle
+					if old != localMd5Hash {
+						return false
+					}
+
+					return true
+				},
+			},
+		},
+		UseJSONNumber: true,
+	}
+}
+
+func resourceApigeeApiCreate(d *schema.ResourceData, meta interface{}) error {
+	ctx := context.TODO()
+	tflog.Info(ctx, "resourceApigeeApiCreate")
+	log.Printf("[DEBUG] resourceApigeeApiCreate")
+
+	log.Printf("[DEBUG] resourceApigeeApiCreate, name=			 	%s", d.Get("name").(string))
+	log.Printf("[DEBUG] resourceApigeeApiCreate, org_id=, 			%s", d.Get("org_id").(string))
+	log.Printf("[DEBUG] resourceApigeeApiCreate, config_bundle=, 	%s", d.Get("config_bundle").(string))
+
+	config := meta.(*transport_tpg.Config)
+	userAgent, err := tpgresource.GenerateUserAgentString(d, config.UserAgent)
+	if err != nil {
+		return err
+	}
+
+	var file *os.File
+	var localMd5Hash string
+	if configBundlePath, ok := d.GetOk("config_bundle"); ok {
+		var err error
+		file, err = os.Open(configBundlePath.(string))
+		if err != nil {
+			return err
+		}
+		localMd5Hash = tpgresource.GetFileMd5Hash(configBundlePath.(string))
+	} else {
+		return fmt.Errorf("Error, \"config_bundle\" must be specified")
+	}
+
+	url, err := tpgresource.ReplaceVars(d, config, "{{ApigeeBasePath}}organizations/{{org_id}}/apis?name={{name}}&action=import")
+	if err != nil {
+		return err
+	}
+	billingProject := ""
+
+	// err == nil indicates that the billing_project value was found
+	if bp, err := tpgresource.GetBillingProject(d, config); err == nil {
+		billingProject = bp
+	}
+
+	log.Printf("[DEBUG] resourceApigeeApiCreate, url=, 	%s", url)
+	res, err := sendRequestRawBodyWithTimeout(config, "POST", billingProject, url, userAgent, file, "application/octet-stream", d.Timeout(schema.TimeoutCreate))
+
+	log.Printf("[DEBUG] sendRequestRawBodyWithTimeout Done")
+	if err != nil {
+		return fmt.Errorf("Error creating API proxy: %s", err)
+	}
+
+	// Store the ID now
+	id, err := tpgresource.ReplaceVars(d, config, "organizations/{{org_id}}/apis/{{name}}")
+	if err != nil {
+		return fmt.Errorf("Error constructing id: %s", err)
+	}
+	d.SetId(id)
+	log.Printf("[DEBUG] create d.SetId done, id = %s", id)
+
+	log.Printf("[DEBUG] Finished creating API proxy %q: %#v", d.Id(), res)
+
+	if resourceApigeeApiRead(d, meta) != nil {
+		return fmt.Errorf("Error reading API proxy at end of Create: %s", err)
+	}
+	d.Set("md5hash", localMd5Hash)
```

Comment:
Why do we set these *after* the read method is called? If they were set during read they would be overwritten here